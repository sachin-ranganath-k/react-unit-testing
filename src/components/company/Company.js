import React, { useState } from "react";

const Company = ({ name }) => {
  const [country, setCountry] = useState("India");

  const countryChangeHandler = (event) => {
    console.log(event);
    setCountry(event.target.value);
  };

  return (
    <div>
      <h2>
        Company | Testing props, Events, States, Hooks, Props, CSS, JSX,
        Snapshot
      </h2>
      <h3 data-testid="companyName">cognizant's official website</h3>
      <h4 data-testid="companyFullName">Company name : {name}</h4>

      <br />
      {/* Working with onchange */}
      <label>Enter your country :</label>
      <input
        type="text"
        placeholder="Enter Country"
        value={country}
        onChange={countryChangeHandler}
      />
      <p>Head over to Cognizant {country} website</p>
    </div>
  );
};

export default Company;
