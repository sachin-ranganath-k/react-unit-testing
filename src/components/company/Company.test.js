import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Company from "./Company";

describe("Company component", () => {
  it("Has company name cognizant", () => {
    render(<Company />);
    const companyNameEl = screen.getByTestId("companyName");
    expect(companyNameEl.textContent).toBe(`cognizant's official website`);
  });

  //Props testing
  it("receives name prop", () => {
    render(<Company name="Cognizant Technology Solutions" />);
    expect(screen.getByTestId("companyFullName")).toHaveTextContent(
      "Cognizant Technology Solutions"
    );
  });

  //input element with 'Enter country' as placeholder
  it(`has an input element with 'Enter country' as placeholder`, () => {
    render(<Company />);
    const country = screen.getByPlaceholderText("Enter Country");
    expect(country).toBeTruthy();
  });

  //useState, onChange testing
  it("should update country onChange event", () => {
    render(<Company />);
    // let's find the input element that has onChange handler
    const countryInput = screen.getByPlaceholderText("Enter Country");

    // now checking if onChange event is working well or not
    // mock fire the onChange event with mock value

    fireEvent.change(countryInput, {
      target: {
        value: "Australia",
      },
    });
    expect(countryInput.value).toBe("Australia");
    expect(screen.getByText("Head over to Cognizant Australia website"));

    fireEvent.change(countryInput, {
      target: {
        value: "UK",
      },
    });
    expect(countryInput.value).toBe("UK");
    expect(screen.getByText("Head over to Cognizant UK website"));
  });

  //Testing inline css
  it("css style with bg color green", () => {
    render(<Company />);
    expect(screen.getByTestId("successNotification")).toHaveStyle(
      "background-color:#00ff00"
    );
  });

  //testing element with css class
  it("css style with class", () => {
    render(<Company />);
    expect(screen.getByTestId("successNotification")).toHaveClass(
      "notification"
    );
  });

  //testing button css with class
  it("css style with class of button", () => {
    render(<Company />);
    expect(screen.getByRole("buttonClick")).toHaveClass("btn-danger");
  });

  //snapshot testing
  it("snapshot testing", () => {
    //npm i react-test-renderer
    const snapshotJSON = renderer.create(<Company name="Cognizant Technology Solutions" />);
    expect(snapshotJSON).toMatchSnapshot();
  });
});
