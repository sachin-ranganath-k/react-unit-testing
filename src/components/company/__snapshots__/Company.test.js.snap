// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company component  1`] = `
<div>
  <h2>
    Company | Testing props, Events, States, Hooks, Props, CSS, JSX, Snapshot
  </h2>
  <h3
    data-testid="companyName"
  >
    cognizant's official website
  </h3>
  <h4
    data-testid="companyFullName"
  >
    Company name : 
    Cognizant Technology Solutions
  </h4>
  <br />
  <label>
    Enter your country :
  </label>
  <input
    onChange={[Function]}
    placeholder="Enter Country"
    type="text"
    value="India"
  />
  <p>
    Head over to Cognizant 
    India
     website
  </p>
  <br />
  <div
    className="notification"
    data-testid="successNotification"
    style={
      Object {
        "backgroundColor": "#00ff00",
        "padding": "10px",
      }
    }
  >
    Data Saved...!
  </div>
  <button
    className="btn-danger"
    role="buttonClick"
  >
    Danger Button
  </button>
</div>
`;

exports[`Company component snapshot testing 1`] = `
<div>
  <h2>
    Company | Testing props, Events, States, Hooks, Props, CSS, JSX, Snapshot
  </h2>
  <h3
    data-testid="companyName"
  >
    cognizant's official website
  </h3>
  <h4
    data-testid="companyFullName"
  >
    Company name : 
    Cognizant Technology Solutions
  </h4>
  <br />
  <label>
    Enter your country :
  </label>
  <input
    onChange={[Function]}
    placeholder="Enter Country"
    type="text"
    value="India"
  />
  <p>
    Head over to Cognizant 
    India
     website
  </p>
  <br />
  <div
    className="notification"
    data-testid="successNotification"
    style={
      Object {
        "backgroundColor": "#00ff00",
        "padding": "10px",
      }
    }
  >
    Data Saved...!
  </div>
  <button
    className="btn-danger"
    role="buttonClick"
  >
    Danger Button
  </button>
</div>
`;
